src/imitation/rewards/discrim_net.py

```diff
@@ -273,26 +273,72 @@ def _load(cls, directory):
     return cls(reward_net=reward_net, **params)
 
 
+DiscrimNetBuilder = Callable[[tf.Tensor, tf.Tensor],
+                             Tuple[util.LayersDict, tf.Tensor]]
+"""Type alias for function that takes an observation and action tensor and
+produces a tuple containing (1) a list of used TF layers, and (2) output logits
+for GAIL."""
+
+
+class build_mlp_discrim_net:
```

Comment:
What sort of changes do we need to add in order to allow `Sacred` to modify the hidden sizes? Prior to this PR, it's possible via some very nested dictionary definitions like `init_trainer_kwargs....hidden_sizes = (32, 32)`.

Oh I suppose that we could later add a `hidden_sizes` top-level config to `scripts.config.train_adversarial`, and then have `scripts.config.train_adversarial` automatically convert that to a `build_mlp_discrim_net`.

Reply:
Yep, I think Sacred-friendly stuff should be pushed to `train_adversarial()`; anyone using these internal classes will probably want more direct control.

Reply:
Oh, I take it back; I can just let the user pass extra kwargs for `build_discrim_net` straight in. This doesn't fully solve the problem of specifying networks from the command line (e.g. what if you want to replace CNN with MLP?), but I'm going to declare that as out of scope for this PR.

Reply:
>  can just let the user pass extra kwargs for build_discrim_net straight in.

If it's fine with you, I can make this change before we merge this PR.

Reply:
I already did change it, see https://github.com/HumanCompatibleAI/imitation/pull/133/files/#diff-c976e486405c3d9c2bb8d4bbd2aac4f1R315

Reply:
Oh yeah, just saw that thanks